# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../honeyd/gcc4.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -Nur honeyd-1.0a-rc2.orig/dhcpclient.c honeyd-1.0a-rc2/dhcpclient.c
--- honeyd-1.0a-rc2.orig/dhcpclient.c	2005-04-07 09:31:49.000000000 +0300
+++ honeyd-1.0a-rc2/dhcpclient.c	2005-11-01 14:50:05.000000000 +0200
@@ -472,7 +472,8 @@
 
 	memset(buf, 0, sizeof(buf));
 
-	eth = (struct eth_hdr *)p = buf;
+	p = buf;
+	eth = (struct eth_hdr *)p;
 	eth_pack_hdr(eth, ETH_ADDR_BROADCAST, req->ea, ETH_TYPE_IP);
 
 	restlen -= ETH_HDR_LEN;
@@ -524,7 +525,8 @@
 
 	memset(buf, 0, sizeof(buf));
 
-	eth = (struct eth_hdr *)p = buf;
+	p = buf;
+	eth = (struct eth_hdr *)p;
 	eth_pack_hdr(eth, req->server_ea, req->ea, ETH_TYPE_IP);
 
 	restlen -= ETH_HDR_LEN;
